### PR TITLE
issue 1953 - added getOptPath function to tuneParams.R

### PR DESCRIPTION
**Pull request by scholbeck**
- a getOptPath() function was added to tuneParams.R
- it is a getter method for obtaining the optimization path from a tuning result
- it provides the option to print the optimization path as a data frame
- added a unit test `test_tune_getTuneResultOptPath.R`

### CODE REVIEW

**Comment by scholbeck** [comment]
fix for #1953

**Comment by larskotthoff** [comment]
Could you add a unit test for this please?

**Comment by scholbeck** [comment]
Yes. I'm going to add a unit test.

**Comment by jakob-r** [comment]
Following the mlr naming convention this function should be called `getTuneResultOptPath`.
If we want to stick with this naming it has to be S3.

File: R/tuneParams.R [hunk 1]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
 }
 
 
+#' @title Get the optimization path of a tuning result.
+#'
+#' @description
+#' Returns the optPath from a [\code{\link{tuneParams}}] object.
+#' @param tune.result [\code{\link{tuneParams}}] \cr
```
Comment (jakob-r) [inline R/tuneParams.R]: has to be `TuneResult`.

File: R/tuneParams.R [hunk 2]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
...
+#'
+#' @description
+#' Returns the opt.path from a [\code{\link{tuneParams}}] object.
+#' @param TuneResult [\code{\link{tuneParams}}] \cr
+#'   A tuning result of the [\code{\link{tuneParams}}] function.
+#' @param as.df [\code{logical(1)}]\cr
+#'   Should the optimization path be returned as a data frame?
+#'   Default is \code{TRUE}.
+#' @export
+getTuneResultOptPath = function(TuneResult, as.df = TRUE)
```
Comment (jakob-r) [inline R/tuneParams.R]: `tune.result`

File: R/tuneParams.R [hunk 3]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
 }
 
 
+#' @title Get the optimization path of a tuning result.
+#'
+#' @description
+#' Returns the opt.path from a [\code{\link{tuneParams}}] object.
+#' @param TuneResult [\code{\link{tuneParams}}] \cr
```
Comment (jakob-r) [inline R/tuneParams.R]: TuneResult -> `tune.result`
tuneParams -> `TuneResult`

File: R/tuneParams.R [hunk 4]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
 }
 
 
+#' @title Get the optimization path of a tuning result.
+#'
+#' @description
+#' Returns the opt.path from a [\code{\link{tuneParams}}] object.
```
Comment (jakob-r) [inline R/tuneParams.R]: tuneParams -> `TuneResult`

File: R/tuneParams.R [hunk 5]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
...
+#' @param TuneResult [\code{\link{tuneParams}}] \cr
+#'   A tuning result of the [\code{\link{tuneParams}}] function.
```
Comment (jakob-r) [inline R/tuneParams.R]: tuneParams -> `TuneResult`

Comment (scholbeck) [reply R/tuneParams.R]: Are you sure about this one? I just want to show the user in line 135, that the tuning result is created by the tuneParams function.

**Comment by scholbeck** [comment]
I'm sorry about that. I'm new to mlr and couldn't find the naming conventions in the styleguide.

File: R/tuneParams.R (hunk 5 above)
Comment (jakob-r) [reply R/tuneParams.R]: You are right L:135 it's the result of tuneParams. But in L:134 it has to be TuneResult because that is the class of the object

**Comment by jakob-r** [comment]
Everything looks fine now. Except one last request. I would prefer if we don't have just another tuning call only to test this simple function. Could be incorporated to other tests.

@larskotthoff What do you think?

File: R/tuneParams.R [hunk 6]
```diff
@@ -127,3 +127,21 @@ tuneParams = function(learner, task, resampling, measures, par.set, control, sho
...
+#' @description
+#' Returns the opt.path from a [\code{\link{TuneResult}}] object.
+#' @param tune.result [\code{\link{TuneResult}}] \cr
+#'   A tuning result of the [\code{\link{tuneParams}}] function.
+#' @param as.df [\code{logical(1)}]\cr
+#'   Should the optimization path be returned as a data frame?
+#'   Default is \code{TRUE}.
+#' @export
+getTuneResultOptPath = function(tune.result, as.df = TRUE)
+{
```
Comment (larskotthoff) [inline R/tuneParams.R]: Opening { should be on the same line as the function declaration.

**Comment by larskotthoff** [comment]
Looks good to me apart from the one comment I made. I'm fine with the test code as it is.

**Comment by scholbeck** [comment]
Sorry about that. Fixed it.

**Comment by larskotthoff** [comment]
And a space in between the ( and { :)

**Comment by jakob-r** [comment]
I still think the test is quite "expensive" but ok...

**Comment by larskotthoff** [comment]
It doesn't look like this is increasing build time, so I'll merge it. We can always change it later.

**Pull request merged**